client/src/components/OpportunitiesList.js

```diff
@@ -193,40 +209,85 @@ class OpportunitiesList extends Component {
         </Divider>
         <br />
         <Grid>
-          {this.state.OpportunitiesList.map((opportunity, index) => (
-            <Card
-              centered
-              key={index}
-              raised
-              color="blue"
-              onClick={this.handleSelectOpportunity}
-              name={opportunity.opportunity_id}
-            >
-              <Card.Content>
-                <Image
-                  floated="right"
-                  size="mini"
+          {this.state.filtering ? (

```

Comment:
is there any difference between what you show when filtering or not? if there isn't then you should just do something like `const list = this.state.filtering? this.state.fitleredList : this.state.opportunitiesList` ... and then use the map once over `list` ... otherwise a lot of the code is duplicated.